main.go

```diff
@@ -92,20 +94,10 @@ func main() {
 		os.Exit(1)
 	}
 
-	catsrcReconciler := controllers.NewCatalogSourceReconciler(
-		mgr.GetClient(),
-		mgr.GetScheme(),
-		mgr.GetEventRecorderFor("catalogsource-controller"),
-	)
-	if err := catsrcReconciler.SetupWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create catalog source controller", "controller", "CatalogSource")
-		os.Exit(1)
-	}
-
 	if err = (&controllers.OperatorReconciler{
 		Client:   mgr.GetClient(),
 		Scheme:   mgr.GetScheme(),
-		Resolver: resolution.NewOperatorResolver(mgr.GetClient(), catsrcReconciler),
+		Resolver: resolution.NewOperatorResolver(mgr.GetClient(), entitysources.NewCatalogdEntitySource(context.TODO(), mgr.GetClient())),
```

Comment:
would it make sense to use the same context (or child context) as the manager `ctrl.SetupSignalHandler()` ?

Reply:
Thank you! Wasn't sure which context was available to us here 🙏🏽

Reply:
Actually, looks like that function has some dealings with a channel that gets closed when you call that function, and trying to use that context variable twice just leads to: 

```
2023-04-27T20:07:30Z	INFO	controller-runtime.metrics	Metrics server is starting to listen	{"addr": "127.0.0.1:8080"}
2023-04-27T20:07:30Z	INFO	setup	starting manager
panic: close of closed channel

goroutine 1 [running]:
sigs.k8s.io/controller-runtime/pkg/manager/signals.SetupSignalHandler()
	/Users/anbhatta/go/src/github.com/operator-framework/operator-controller/vendor/sigs.k8s.io/controller-runtime/pkg/manager/signals/signal.go:31 +0x25
main.main()
	/Users/anbhatta/go/src/github.com/operator-framework/operator-controller/main.go:116 +0x7cc
```

I'm just reverting this to context.TODO() for now and we can definitely look into it later.

Reply:
Oh wait, we don't need this anymore anyway https://github.com/operator-framework/operator-controller/pull/182#discussion_r1179603532 XD